strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -464,28 +264,25 @@ public void deleteMetadata(String storageId,
 
         Storage storage = getConfiguration().getStorage(storageId);
         Repository repository = storage.getRepository(repositoryId);
-        StorageProvider storageProvider = storageProviderRegistry.getProvider(repository.getImplementation());
-
-        final File repoPath = storageProvider.getFileImplementation(repository.getBasedir());
-
+        RepositoryPath repositoryPath = resolve(repository);
+        if (!Files.isDirectory(repositoryPath))
```

Comment:
If `repositoryPath` is not a directory it just do nothing and returns. So it is specifically for directories as originally.